### PR TITLE
bin, funds-manager: Add deployment and upgrade scripts

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR adds a `Dockerfile` to the funds manager and generalizes the existing `build_and_push.sh` and `deploy.sh` scripts to work for all extensions.

 ### Testing
- Deployed to testnet

### CODE REVIEW

File: bin/build_and_push.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+REGION=us-east-2
```
Comment (akirillo) [inline bin/build_and_push.sh:2]: this should probably be configurable via command line args as well for e.g. price reporter extension

File: bin/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+set -e
+
+REGION=us-east-2
```
Comment (akirillo) [inline bin/deploy.sh:4]: same here, region should prob be configurable via CLI

File: bin/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+# Derive values from environment and resource
+CLUSTER_NAME="$ENVIRONMENT-$RESOURCE-cluster"
+SERVICE_NAME="$ENVIRONMENT-$RESOURCE-service"
+TASK_DEF_NAME="$ENVIRONMENT-$RESOURCE-task-def"
+ECR_REPO="$RESOURCE-$ENVIRONMENT"
+
+# Fetch the latest image URI from ECR
+ECR_URL="$ECR_REGISTRY/$ECR_REPO"
+IMAGE_URI=$(aws ecr describe-images --repository-name $ECR_REPO --region $REGION --query 'sort_by(imageDetails,& imagePushedAt)[-1].imageTags[0]' --output text)
```
Comment (akirillo) [inline bin/deploy.sh:31]: empirically, this has often given me the `:latest`-tagged image URI, as opposed to the `COMMIT_HASH`-tagged image URI. This is unsafe to put in the task definition as new tasks getting started will use whatever image is tagged w/ latest at launch time. I'd propose we just generate `COMMIT_HASH` as we do in `build_and_push.sh` and use that to construct the `FULL_IMAGE_URI`

**Review by akirillo — APPROVED** [review]

File: bin/deploy.sh (hunk 3 above)
Comment (joeykraut) [reply bin/deploy.sh:31]: done

**Pull request merged**